docs/meetings/exeter-jun-2019.md

```diff
@@ -0,0 +1,254 @@
+
+- OS agrees with DS PRs in large part; GraphQL schema looks good.
+  - not convinced on protobuf yet (but see below)
+
+- "stop suite" should report to user "waiting for active tasks to finish" (e.g.)
+
+- GraphQL has typing too (like Protobuf); not sure  if validation/checking is done
+  - BK wants types in the UI for safety/robustness
+
+- FUTURE: display multiple suites together with inter-suite triggers!
+  - great for small modular workflows
+
+- DS PRs: push back functional and unit tests, and "granularise data
+  structure", to follow-up devel.
+
+- TODO: consider Travis CI cron jobs for cross-project integration testing.
+  - use cylc/cylc-admin for this (T-CI config can pull from other projects)
+
+- tests should run against master all the time, not release versions
+  - need setup.py to install from branch URL not PyPi
+
+- TODO (HO) - check JOSS paper repo link.
+
+- SMS-like ghost nodes in all views? is a broken concept because it elevates
+    "cycle points" to be more fundamental than they will be in Cylc 9 - some
+    workflows might iterate over parameters first (or just wherever the graph leads).
+   - we just need to educate users on the strengths of the cylc "adaptive
+     window on an infinite graph" world view.
+         
+- What to display in the new UI:
+   - first, as now, just display the task pool
+   - then consider a new HYBRID MODE that looks a bit like spawn-on-demand will:
+      - display N-edges out from "active tasks", in all views.
+      - at start-up request N edges out, populate from suite and DB (only need to
+        connect to DB at start-up)
+      - "data display" technology if needed later (??Que??)
+      - "level 0" is "active tasks" (submitted, running, satisfied, ... if
+        prereqs are met OR "cannot be met")
+      - allow configurable n-level zoom
+      - max-n could be determined by query time on page
+      - TODO ask DS: are n=1,2,3 edges already paged?
+      - max tasks permitted per page?
+   - UIS can query the DB to populate ghost nodes (and in future for "exploring
+     the graph" beyond active tasks)
+   - use distinct edge style to indicate that manual intervention forced
+     override of prerequisite status.
+   - failed tasks should need to be "dismissed" (not "removed"!)
+   - show a separate list of failed tasks that have not been dismissed yet
+   - failed tasks should not be treated differently to succeeded in terms of
+     workflow
+       - instead of keeping them around in the task pool should we have
+          a different failure alerting system?
+        - need to distinguish tasks that are expected to fail
+           - FUTURE: generalise this to allow users to specify "expected
+           completion state" of all tasks (default to expect success)
+   - NOTE two kinds of ghost nodes!
+      1) graph node with no corresponding task proxy currently in the pool
+         (this kind will cease to exist in the spawn-on-demand future).
```

Comment:
Nodes N-generations away from active tasks. Can be N-parents or N-children. Should these be ghost tasks? Or just call them inactive or non-current tasks?

Reply:
Good question.  In the old/current meaning of "ghost" (no task proxy in the pool) some of those will be ghosts and some won't.  So now you want to overload the term again to mean non-active tasks visible when you zoom out in the active window ... he heh  :rofl: :rofl:

Hmm, seriously, I think just "active" and "inactive" will do in the future. Ghost implies something a bit mysterious, which "not present in the task pool" is.  But "active or not" should be much less mysterious to users, I hope.

Reply:
Noted.